documentation/blog.md

```diff
@@ -0,0 +1,27 @@
+As you already know, YaST was converted from YCP to Ruby some time ago. At least we repeat it quite often ;-) However, this conversion was done on language basis. Some old design decissions and principles stayed. One of them is usage of SCR for accessing underlying configuration files.
+
+SCR was designed together with YaST. It uses concept of "agents" for accessing configuration files. These agents contains a description of configuration file using own format. Moreover SCR offers location transparency. You can e.g. work with a file in system or in a chrooted environment. However, this piece of code is proprietaty and limited by different quality of agents. It is also written in C++ and development is done only in SUSE. From this and some other reasons we decided to replace proprietary SCR with something else. We started to use "Config files API"
+
+Config files API (CFA) is a library written in ruby intended for accessing various configuration files. It is structured into several layers and creates an internal abstraction of configuration file. It has again been designed and developed in SUSE's YaST team. However this time it uses (or can use) third party parsers. CFA provides location transparency for the parser on the bottom layer and unific API for application on the top one.
```

Comment:
maybe mention that location transparency is done by well known `File` interface.